src/main/kotlin/io/openfuture/state/webhook/DefaultWebhookExecutor.kt

```diff
@@ -0,0 +1,43 @@
+package io.openfuture.state.webhook
+
+import io.openfuture.state.config.WebhookConfig
+import io.openfuture.state.domain.WebhookInvocation
+import io.openfuture.state.util.MathUtils
+import kotlinx.coroutines.delay
+import org.springframework.stereotype.Service
+import org.springframework.web.client.RestTemplate
+
+@Service
+class DefaultWebhookExecutor(
+        private val webhookConfig: WebhookConfig
+): WebhookExecutor {
+
+    override suspend fun execute(webhookInvocation: WebhookInvocation) {
+        try {
+            for (i in 1..webhookConfig.maxAttempts()) {
+
+                webhookInvocation.attempts++
+                if (executeGetRequest(webhookInvocation.url)) {
+                    webhookInvocation.status = WebhookStatus.OK
```

Comment:
Is it a good practice to change the state of the object in a execute function? I think it should just execute and as a execution result should return its own contract on which caller will work